server/base/src/main/java/org/apache/accumulo/server/metadata/GcMutatorImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.server.metadata;
+
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.fs.FileRef;
+import org.apache.accumulo.server.metadata.MetadataMutator.GcMutator;
+import org.apache.accumulo.server.util.MetadataTableUtil;
+
+public class GcMutatorImpl implements GcMutator {
+
+  private ServerContext context;
+  private TableId tableId;
+
+  public GcMutatorImpl(ServerContext context, TableId tableId) {
```

Comment:
An early suggestion: whenever TableId (or tableName) is used in these, would be good to add a comment to explain whether this is the table whose metadata is being mutated, or if it's the metadata table in which the metadata is being mutated.